### PR TITLE
[Network] Network Watcher fixes

### PR DESCRIPTION
**Pull request by tjprescott**
Adds `network watcher test-connectivity` command.
Adds support for `--filters` to `network watcher packet-capture create`.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.  Cannot add test because service endpoint is not live. Have tested manually on service's staging subscription. (#3142)

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by tjprescott** [comment]
```
Command
    az network watcher test-connectivity: Test whether a direct TCP connection can be established
    between a Virtual Machine and a given endpoint, such as another VM or an arbitrary remote
    server.

Arguments
    --source-resource [Required]: Name or ID of the resource from which to originate traffic.
        Currently only Virtual Machines are supported.
    --resource-group -g         : Name of resource group. You can configure the default group using
                                  `az configure --defaults group=<name>`.
    --source-port               : Port number from which to originate traffic.

Destination Arguments
    --dest-address              : The IP address or URI at which to receive traffic.
    --dest-port                 : Port number on which to receive traffic.
    --dest-resource             : Name or ID of the resource to receive traffic.
        Currently only Virtual Machines are supported.
```

**Comment by tjprescott** [comment]
Added DNM label since this requires the SDK to release an update that supports the new command. Currently in PR. 
https://github.com/AutorestCI/azure-sdk-for-python/tree/RestAPI-PR1175/azure-mgmt-network

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py [hunk 1]
```diff
@@ -661,10 +661,39 @@ def _validator(namespace):
...
+    compute_client = get_mgmt_service_client(ResourceType.MGMT_COMPUTE).virtual_machines
+    vm_name = parse_resource_id(namespace.source_resource)['name']
+    rg = namespace.resource_group_name or \
+        parse_resource_id(namespace.source_resource).get('resource_group', None)
+    if not rg:
+        raise CLIError('usage error: --source-resource ID | '
+                       '--source-resource NAME --resource-group NAME')
+    vm = compute_client.get(rg, vm_name)
+    namespace.location = vm.location  # pylint: disable=no-member
+    get_network_watcher_from_location(remove=True)(namespace)
```
Comment (troydai) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py]: I found it a bit odd to use a `get` function to remove a property from the namespace.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py [hunk 2]
```diff
@@ -661,10 +661,39 @@ def _validator(namespace):
...
+    if namespace.dest_resource and not is_valid_resource_id(namespace.dest_resource):
+        namespace.dest_resource = resource_id(
+            subscription=get_subscription_id(),
+            resource_group=namespace.resource_group_name,
+            namespace='Microsoft.Compute',
+            type='virtualMachines',
+            name=namespace.dest_resource)
+
+def process_nw_flow_log_set_namespace(namespace):
+
```
Comment (troydai) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py]: Unnecessary blank line.

**Comment by troydai** [comment]
It would be preferable to have a test.

**Comment by tjprescott** [comment]
@troydai I agree but I can't have a test until the functionality is released.  I've got a work item in Sprint 16 to add this test. #3142.

**Pull request merged**